packages/core/admin/webpack.config.js

```diff
@@ -204,7 +206,8 @@ module.exports = ({
       new webpack.DefinePlugin(envVariables),
 
       new NodePolyfillPlugin(),
+      !isProduction && process.env.REACT_REFRESH !== 'false' && new ReactRefreshWebpackPlugin(),
```

Comment:
If this PR gets merged we will need to add some documentation.

cc/ @pwizla & @StrapiShaun

Reply:
It looks really cool. Just let us know!

Reply:
Can you guys help with the documentation? I'm not very familiar with it.

Reply:
Sure @nvh95, and thanks for the mention @soupette .
Basically it would just mean that we need to add a line to [this table](https://docs.strapi.io/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.html#strapi-s-environment-variables) as this PR would add a new REACT_REFRESH (boolean) environment variable.

If you feel like creating the documentation PR yourself, @nvh95, here's our [contributing guide](https://github.com/strapi/documentation/blob/main/CONTRIBUTING.md#submitting-a-pull-request). You'd have to create a PR targeting the `main` branch. [Here](https://github.com/strapi/documentation/blob/main/docs/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md) is the file that needs to be updated.

Reply:
@pwizla I would love to. But recently I don't have much time to spend on strapi. I would like to ask you to help me, if it's possible. In the next PRs/ features, I definitely update the documentation follow the guidelines as well.
Thank you.

Reply:
That's totally OK, @nvh95 ! I'll create the documentation PR and ask you to review it, to make sure I properly describe what you had in mind with this environment variable. Sounds good to you? 🙂

Reply:
@pwizla absolutely. please mention @nvh95. Thanks.